packages/twenty-server/src/engine/metadata-modules/remote-server/remote-server.entity.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  Generated,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+
+export enum RemoteServerType {
+  POSTGRES_FDW = 'postgres_fdw',
+}
+
+type PostgresFdwOptions = {
+  host: string;
+  port: number;
+  dbname: string;
+};
+
```

Comment:
I was thinking we could do something like this eventually? (not tested)

```
export enum RemoteServerType {
  POSTGRES = 'postgres_fdw',
  STRIPE = 'stripe_fdw',
}

type PostgresFdwOptions = {
  host: string;
  port: number;
  dbname: string;
};

type StripeFdwOptions = {
  apiKey: string;
};

type FdwOptions<T extends RemoteServerType> = T extends RemoteServerType.POSTGRES
  ? PostgresFdwOptions
  : T extends RemoteServerType.STRIPE
  ? StripeFdwOptions
  : never;
  ```

Reply:
I implemented this. But I did not introduced Stripe yet :)